src/eval.jl

```diff
@@ -20,6 +20,17 @@ function getmodule(data, pos)
   getthing("$main.$sub", getthing(main, Main))
 end
 
+function ends_with_semicolon(line)
+    match = rsearch(line, ';')
+    if match != 0
+        for c in line[(match+1):end]
+            isspace(c) || return c == '#'
+        end
+        return true
+    end
+    return false
+end
```

Comment:
It is, I think. But needs a bit of tweaking anyways -- the base method doesn't handle linebreaks and stuff while we need to.